sed/binning.py

```diff
@@ -405,37 +483,56 @@ def numba_histogramdd(
         N, D = sample.shape
 
     try:
-        M = len(bins)
-        if M != D:
-            raise ValueError(
-                "The dimension of bins must be equal to the dimension of the "
-                " sample x.",
-            )
-    except TypeError:
-        # bins is an integer
+        Db = len(bins)
+        if isinstance(bins[0], int):
```

Comment:
would be simpler if `bins` are always either `ints` or `arrays`??

Reply:
Not sure what you mean...

Reply:
why is it necessary to support either bins or arrays?
Would not be easier to support only arrays?

Reply:
we have two binning methods now, implemented with numba. One based on defining the number of bins (therefore int) and is much faster. The other takes a pre-made array and is slower, but allows non-linear binning.

Therefore the two different input methods. Also, this mimics what numpy.histogramdd does, where you can pass an int or an array.